src/components/Tables/Table.tsx

```diff
@@ -89,41 +125,119 @@ const Table: React.FC<Props> = (props) => {
-                onClick={toggleAllCheckBox}
-            />
-        ),
-        cell: (row: Row) => (
-            <input
-                type="checkbox"
-                aria-label={`Select row ${row.rowId}`}
-                checked={selectCheckBoxes[row.rowId]}
-                onClick={() => toggleOwnCheckBox(row.rowId)}
-            />
-        ),
-        width: "47px",
-    });
+        const newData = [...data];
+        const temp = newData[rowId1];
+        newData[rowId1] = newData[rowId2];
+        newData[rowId2] = temp;
+
+        setData(newData);
+    };
+
+    if (props.checkboxes) {
+        columns.unshift({
+            name: (
+                <input
+                    type="checkbox"
+                    aria-label="Select all rows"
+                    checked={selectAllCheckBox}
+                    onClick={toggleAllCheckBox}
+                />
+            ),
+            cell: (row: Row) => (
+                <input
+                    type="checkbox"
+                    aria-label={`Select row ${row.rowId}`}
+                    checked={selectCheckBoxes[row.rowId]}
+                    onClick={() => toggleOwnCheckBox(row.rowId)}
+                />
+            ),
+            width: "47px",
+        });
+    }
+
+    if (props.reorderable || props.onEdit) {
+        columns.unshift({
+            name: <p>Sort</p>,
+            cell: (row: Row) => {
+                const onEditClick = (): void => {
+                    props.onEdit!(row.rowId);
+                };
+
+                return (
+                    <EditandReorderArrowDiv>
+                        {props.reorderable ? (
+                            <IconButton
+                                onClick={() => swapRows(row.rowId, row.rowId - 1)}
+                                aria-label="reorder row upwards"
+                            >
+                                <StyledIcon icon={faAnglesUp} />
+                            </IconButton>
+                        ) : (
+                            <></>
+                        )}
+                        {props.onEdit ? (
+                            <IconButton onClick={onEditClick} aria-label="edit">
+                                <StyledIcon icon={faPenToSquare} />
+                            </IconButton>
+                        ) : (
+                            <></>
+                        )}
+                        {props.reorderable ? (
+                            <IconButton
+                                onClick={() => swapRows(row.rowId, row.rowId + 1)}
+                                aria-label="reorder row downwards"
+                            >
+                                <StyledIcon icon={faAnglesDown} />
+                            </IconButton>
+                        ) : (
+                            <></>
+                        )}
+                        {props.onDelete ? (
+                            <IconButton
```

Comment:
same here - how much does our existing Icon component cover?

Reply:
Our Icon component adds styling that isn't necessary when used with a material icon button so it's easier and cleaner to not use it.

Reply:
I would much rather we update the Icon and use it wherever possible (for consistency, and for if we need to change every Icon quickly)